drizzle-orm/src/xata-http/migrator.ts

```diff
@@ -0,0 +1,49 @@
+import type { MigrationConfig } from '~/migrator.ts';
+import { readMigrationFiles } from '~/migrator.ts';
+import { type SQL, sql } from '~/sql/sql.ts';
+import type { XataHttpDatabase } from './driver.ts';
+
+/**
+ * This function reads migrationFolder and execute each unapplied migration and mark it as executed in database
+ *
+ * NOTE: The Xata HTTP driver does not support transactions. This means that if any part of a migration fails,
+ * no rollback will be executed. Currently, you will need to handle unsuccessful migration yourself.
+ * @param db - drizzle db instance
+ * @param config - path to migration folder generated by drizzle-kit
+ */
+export async function migrate<TSchema extends Record<string, unknown>>(db: XataHttpDatabase<TSchema>, config: string | MigrationConfig) {
+  const migrations = readMigrationFiles(config);
+  const migrationsTable = typeof config === 'string' ? '__drizzle_migrations' : config.migrationsTable ?? '__drizzle_migrations';
+  const migrationsSchema = typeof config === 'string' ? 'drizzle' : config.migrationsSchema ?? 'drizzle';
+  const migrationTableCreate = sql`
+		CREATE TABLE IF NOT EXISTS ${sql.identifier(migrationsSchema)}.${sql.identifier(migrationsTable)} (
+			id SERIAL PRIMARY KEY,
+			hash text NOT NULL,
+			created_at bigint
+		)
+	`;
+  await db.session.execute(sql`CREATE SCHEMA IF NOT EXISTS ${sql.identifier(migrationsSchema)}`);
+  await db.session.execute(migrationTableCreate);
+
+  const dbMigrations = await db.session.all<{
+    id: number;
+    hash: string;
+    created_at: string;
+  }>(sql`select id, hash, created_at from ${sql.identifier(migrationsSchema)}.${sql.identifier(migrationsTable)} order by created_at desc limit 1`);
+
+  const lastDbMigration = dbMigrations[0];
+  const rowsToInsert: SQL[] = [];
+  for await (const migration of migrations) {
+    if (!lastDbMigration || Number(lastDbMigration.created_at) < migration.folderMillis) {
+      for (const stmt of migration.sql) {
+        await db.session.execute(sql.raw(stmt));
+      }
+
+      rowsToInsert.push(sql`insert into ${sql.identifier(migrationsSchema)}.${sql.identifier(migrationsTable)} ("hash", "created_at") values(${migration.hash}, ${migration.folderMillis})`);
```

Comment:
Depending on the results you want to see here, it may be either a good or a bad decision.

So, currently, here is what will happen:

You have 2 files with migrations (`migration_1.sql`, `migration_2.sql`):

- If the migration process fails after `migration_1.sql` is applied, the row in the `__drizzle_migrations` table won't be created, and you will be in a situation where you have all statements from the first file applied, but nothing is stored about that in the database.

As a user, I would expect the migrator to store information about the applied migration file in the database. Therefore, perhaps you need to run this insert query here and not on lines 46-48

Reply:
I followed the example set by Neon's HTTP migrator: https://github.com/drizzle-team/drizzle-orm/blob/main/drizzle-orm/src/neon-http/migrator.ts#L37-L50

Maybe theirs should also be updated?

Reply:
good point, will need to change it as well, I guess I missed it on a review

Reply:
https://github.com/drizzle-team/drizzle-orm/pull/1902/commits/6ce62ea2c1e2f81568e515bb18395d0324a13a47